gap/ClassicalMaximals.gi

```diff
@@ -997,18 +1015,135 @@ function(n, q)
     fi;
 
     primeDivs := PrimePowersInt(n);
-    if not Length(primeDivs) = 2 then
+    if Length(primeDivs) <> 2 then
         return [];
     fi;
 
-
     listOfts := Filtered(DivisorsInt(primeDivs[2]), IsOddInt);
     RemoveSet(listOfts, 1);
 
+    # This way we avoid (m, q) = (2, 3) according to Table 2.10 in [BHR13]
+    if q = 3 then
+        RemoveSet(listOfts, primeDivs[2]);
+    fi;
+
     return List(listOfts, t -> TensorInducedDecompositionStabilizerInSp(primeDivs[1] ^ QuoInt(primeDivs[2], t), t, q));
 end);
 
 BindGlobal("C8SubgroupsSymplecticGroupGeneric",
 function(n, q)
     return [OrthogonalInSp(1, n, q), OrthogonalInSp(-1, n, q)];
 end);
+
+InstallGlobalFunction(MaximalSubgroupClassRepsSymplecticGroup,
+function(n, q, classes...)
+    local maximalSubgroups;
+
+    if Length(classes) = 0 then
+        classes := [1..9];
+    elif IsList(classes[1]) then
```

Comment:
```suggestion
    elif Length(classes) = 1 and IsList(classes[1]) then
```
So that `MaximalSubgroupClassRepsSymplecticGroup(n, q, [1,2,4], 5)` is not accepted